torch_xla/core/dynamo_bridge.py

```diff
@@ -447,6 +459,33 @@ def optimized_mod(*args: tuple):
     nonlocal dumb_return_handler
     nonlocal xla_args_need_update
     nonlocal skip_checking_input_sharding_threashold
+    nonlocal input_shape_mappings
+
+    if not torch._dynamo.config.assume_static_by_default:
```

Comment:
do we need to cache this value in `extracted_internal`? for example if user do 
```
torch.compile(, dynamic=True)
torch.compile(, dynamic=False)
```
what will happen?

Reply:
That's a good point. According to my testing, it seems like the flag to the first `torch.compile` call determines the value of `torch._dynamo.config.assume_static_by_default` throughout the program.

Ex. 1: `dynamic=False` in the first `torch.compile` call
Code:
```
    compiled_add_fn = torch.compile(fn, backend='openxla', dynamic=False)
    a = torch.randn(3, 4, device=device)
    b = torch.ones(4, device=device)
    c = compiled_add_fn(a, b)
    print(f'{c=}')

    compiled_add_fn_2 = torch.compile(simple_sub, backend='openxla', dynamic=True)
    d = torch.randn(3, 4, device=device)
    e = torch.ones(4, device=device)
    f = compiled_add_fn(d, e)
    print(f'{f=}')

    a2 = torch.randn(3, 4, device=device)
    b2 = torch.ones(4, device=device)
    c2 = compiled_add_fn(a2, b2)
    print(f'{c2=}')
```
Result: the value of `torch._dynamo.config.assume_static_by_default` is False for all 3 `optimized_mod` calls

Ex. 2: `dynamic=True` in the first `torch.compile` call
Code:
```
    compiled_add_fn = torch.compile(fn, backend='openxla', dynamic=True)
    a = torch.randn(3, 4, device=device)
    b = torch.ones(4, device=device)
    c = compiled_add_fn(a, b)
    print(f'{c=}')

    compiled_add_fn_2 = torch.compile(simple_sub, backend='openxla', dynamic=False)
    d = torch.randn(3, 4, device=device)
    e = torch.ones(4, device=device)
    f = compiled_add_fn(d, e)
    print(f'{f=}')

    a2 = torch.randn(3, 4, device=device)
    b2 = torch.ones(4, device=device)
    c2 = compiled_add_fn(a2, b2)
    print(f'{c2=}')
```
Result: the value of `torch._dynamo.config.assume_static_by_default` is True for all 3 `optimized_mod` calls

This looks like the behavior of `torch._dynamo.config`, as the values are initialized and set when the first `torch.compile` is called. I don't expect this to be a common scenario, but I can follow up with Edward in the PyTorch GitHub issue about this.

Reply:
Ok, I found out why my test above is behaving wrong -- it was an error in my test script.

To come back to the original question:
> Do we need to cache this value in extract_internal? For example if user do
> torch.compile(, dynamic=True)
> torch.compile(, dynamic=False)
> What will happen?
We don't need to cache `torch._dynamo.config.assume_static_by_default`. This value is per `torch.compile` call. On your example, the second `torch.compile` will correctly have `torch._dynamo.config.assume_static_by_default=True`.